aapenimservice/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/aapenimservice/AapenImService.kt

```diff
@@ -0,0 +1,281 @@
+//                        Key.AAPEN_INNTEKTMELDING to inntektsmelding.toJson(Inntektsmelding.serializer())
+//                    )
+
+                    // TODO Midlertidig sett svar til im-api
+                    val clientId = redisStore.get(RedisKey.of(transaksjonId, event))!!.let(UUID::fromString)
+                    redisStore.set(RedisKey.of(clientId), inntektsmelding.toJsonStr(Inntektsmelding.serializer()))
+                } else {
+                    "Fant ikke skjema i Redis. Kan ikke fortsette.".also {
+                        logger.error(it)
+                        sikkerLogger.error(it)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun finalize(melding: Map<Key, JsonElement>) {
+        val transaksjonId = Key.UUID.les(UuidSerializer, melding)
+        val aapenId = Key.AAPEN_ID.les(UuidSerializer, melding)
+        val erDuplikat = Key.ER_DUPLIKAT_IM.les(Boolean.serializer(), melding)
+        val inntektsmeldingJson = Key.AAPEN_INNTEKTMELDING.les(JsonElement.serializer(), melding)
+
+        MdcUtils.withLogFields(
+            Log.klasse(this),
+            Log.event(event),
+            Log.transaksjonId(transaksjonId),
+            Log.aapenId(aapenId)
+        ) {
+            val clientId = redisStore.get(RedisKey.of(transaksjonId, event))?.let(UUID::fromString)
+
+            if (clientId == null) {
+                sikkerLogger.error("Forsøkte å fullføre, men clientId mangler i Redis.")
+            } else {
+                redisStore.set(RedisKey.of(clientId), inntektsmeldingJson.toString())
+            }
+
+            if (!erDuplikat) {
+                rapid.publish(
+                    Key.EVENT_NAME to EventName.AAPEN_IM_LAGRET.toJson(),
+                    Key.UUID to transaksjonId.toJson(),
+                    Key.AAPEN_ID to aapenId.toJson(),
+                    Key.AAPEN_INNTEKTMELDING to inntektsmeldingJson
+                )
+                    .also {
+                        MdcUtils.withLogFields(
+                            Log.event(EventName.AAPEN_IM_LAGRET)
+                        ) {
+                            logger.info("Publiserte melding.")
+                            sikkerLogger.info("Publiserte melding:\n${it.toPretty()}")
+                        }
+                    }
+            }
+        }
+    }
+
+    override fun onError(melding: Map<Key, JsonElement>, fail: Fail) {
+        MdcUtils.withLogFields(
+            Log.klasse(this),
+            Log.event(EventName.AAPEN_IM_MOTTATT),
+            Log.transaksjonId(fail.transaksjonId)
+        ) {
+            val utloesendeBehov = Key.BEHOV.lesOrNull(BehovType.serializer(), fail.utloesendeMelding.toMap())
+            if (utloesendeBehov == BehovType.VIRKSOMHET) {
+                val virksomhetKey = RedisKey.of(fail.transaksjonId, Key.VIRKSOMHET)
+                redisStore.set(virksomhetKey, ukjentVirksomhet())
```

Comment:
Er det mulig å inkludere orgnr i ukjentVirksomhet? Orgnavn er jo ikke så viktig uansett, men hvilket orgnr som ble sendt inn, bør være med

Reply:
For slik jeg leser koden, vil step1 defaulte kun hvis ikke noe er satt i redis?

Reply:
Orgnr ligger i skjema-klassen, så det er alltid med.